### ISSUE TITLE
[HOLD for payment 2024-11-13] [$250] Report fields - Initial value field shows disabled list value but it is not shown in the list

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#48429]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v9.0.27-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y 
**Issue reported by:** Applause Internal Team

## Action Performed:
1. Go to staging.new.expensify.com
2. Go to workspace settings > Report fields.
3. Click Add field > Click Type > Select List.
4. Click List values.
5. Add a few list values.
6. Return to the previous page.
7. Click Initial value > Select any list value as default list value.
8. Click List values.
9. Disable all the list values.
10. Save the list type report field.
11. Click on the added list type report field.
12. Click List values.
13. Add a new list value.
14. Return to previous page.
15. Note that Initial value field shows the disabled list value, but when opening the page, the disabled list value is absent.

## Expected Result:
The Initial value field should not display the disabled list value (disabled in Step 9) because the list value is disabled.
The field should be empty while clickable.

## Actual Result:
The Initial value field shows the disabled list value, but it is not shown in the list in Initial value page.

## Workaround:
Unknown

## Platforms:

- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/69cbd33f-34df-4983-b2ac-da745317b87a

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021831343278141946652</li>
        <li>Upwork Job ID: 1831343278141946652</li>
        <li>Last Price Increase: 2024-10-22</li>
<li>Automatic offers: </li>
<ul>
<li>akinwale | Reviewer | 104573420</li>
<li>abzokhattab | Contributor | 104573428</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @CortneyOfstad</details>

**Comment by IuliiaHerets** [comment]
@dylanexpensify  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by etCoderDysto** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Report fields - Initial value field shows disabled list value but it is not shown in the list
### What is the root cause of that problem?
We are not checking if the default value (`reportField.defaultValue`) is disabled before returning `reportField.defaultValue` in `getReportFieldInitialValue` 
https://github.com/Expensify/App/blob/33b331dd5c0c93ba70ebadc7f0944d8226bfb7d5/src/libs/WorkspaceReportFieldUtils.ts#L73-L78
### What changes do you think we should make in order to solve the problem?
We should check if the default value is disabled first, and return empty string if it is disabled 
```javascript
function getReportFieldInitialValue(reportField: PolicyReportField | null): string {
    const indexOfDefaultValue = reportField?.values.indexOf(reportField?.defaultValue);
    const defalutValueDisabled = reportField?.disabledOptions[indexOfDefaultValue];
    if (defalutValueDisabled) {
        return '';
    }
```
### What alternative solutions did you explore? (Optional)

**Comment by etCoderDysto** [comment]
I am able to reproduce this.

**Comment by hayes102** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
In the report fields, the initial value field displays a disabled list value, but this value is not shown in the list.


### What is the root cause of that problem?
inside the `getReportFieldInitialValue ` function we dont check if the value is disabled or not 

https://github.com/Expensify/App/blob/33b331dd5c0c93ba70ebadc7f0944d8226bfb7d5/src/libs/WorkspaceReportFieldUtils.ts#L73-L78

### What changes do you think we should make in order to solve the problem?
We should check if the default value is disabled, then return empty string


```js
function getReportFieldInitialValue(reportField: PolicyReportField | null): string {
    if (!reportField) return ''; // we shouldnt remove this line as done in the first proposal 
const indexOfDefaultValue = reportField?.values?.indexOf(reportField.defaultValue) ?? -1;
const defaultValueDisabled = indexOfDefaultValue >= 0 && reportField?.disabledOptions?.[indexOfDefaultValue];
if (defaultValueDisabled) return ""
 ```

**Comment by dylanexpensify** [comment]
@akinwale can we review proposals? 🙏

**Comment by akinwale** [comment]
We can move forward with @etCoderDysto's [proposal](https://github.com/Expensify/App/issues/48429#issuecomment-2325257830) here.

🎀👀🎀 C+ reviewed.

**Comment by etCoderDysto** [comment]
@MonilBhavsar I am waiting for [assignment ](https://github.com/Expensify/App/issues/48429#issuecomment-2345361271)

Thank you!

**Comment by MonilBhavsar** [comment]
Looking today...

**Comment by MonilBhavsar** [comment]
> while this solution hides the disabled value on the frontend, it doesn't prevent the disabled value from being sent and saved in the backend. As shown in the video, the value 1 is still being transmitted.

@abzokhattab good point. Could you elaborate when are we making an API call? Is it while disabling the rates or setting the default. Actually hard to follow from the video. I think we should prevent disabling default if it's disabled. cc @thienlnam for your thoughts since I know you worked on Report fields

**Comment by thienlnam** [comment]
Ah, this is the setup for NewDot so cc @mountiny for this one.

There shouldn't be a problem with disabling all of them you can do that on OldDot as well, we should just hide the dropdown if there are no valid options

**Comment by mountiny** [comment]
Oof i swear i have responded before. Sorry about that.

I havent done any changes to how this logic works in the backend so i would suspect ideally we confirm where the issue is happening.

@akinwale can you try confirming taking the same steps in
1. newDot
2. OldDot

And then sign out and sign back in and check what is the policy fieldList state. Is it same on both? Is the behaviour same in oldDot?

**Comment by mountiny** [comment]
@abzokhattab thanks for clarifying. And just to be super clear, the way how your changes will change the newdot behaviour will match it with how oldDot works?

**Comment by mountiny** [comment]
nice yeah so happy to go with the proposal that will make it behave the same as in OldDot cc @MonilBhavsar @akinwale I think its clearer now

**Comment by etCoderDysto** [comment]
@abzokhattab can you please upload a video showing how your proposed solution fixes the issue you have mentioned?

**Comment by dylanexpensify** [comment]
bump @akinwale

**Comment by dylanexpensify** [comment]
Reassigning as I head to parental leave next week! Thanks Cortney!

**Comment by rezkiy37** [comment]
Hi, I am Michael (Mykhailo) from Callstack, an expert agency and I can work on this issue.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by rezkiy37** [comment]
> The issue is not assigned to me yet. The pr will be ready once assigned

Bump @dylanexpensify

**Comment by CortneyOfstad** [comment]
Sorry, was OoO so going to get this adjusted!

@abzokhattab assigning you now!

**Comment by CortneyOfstad** [comment]
Payment set for tomorrow!

**Comment by CortneyOfstad** [comment]
## Payment Summary

@akinwale — paid $250 via Upwork
@abzokhattab — paid $250 via Upwork
